src/frontend/admin/src/components/presentational/hook-form/RHFSelect.tsx

```diff
@@ -1,22 +1,54 @@
 import { Controller, useFormContext } from "react-hook-form";
 import TextField, { TextFieldProps } from "@mui/material/TextField";
+import * as React from "react";
+import { ReactNode } from "react";
+import MenuItem from "@mui/material/MenuItem";
+import Box from "@mui/material/Box";
+import InputAdornment from "@mui/material/InputAdornment";
+import Divider from "@mui/material/Divider";
+import { Maybe } from "@/types/utils";
+
+export interface SelectOption {
+  label: string;
+  value: any;
+}
 
 type RHFSelectProps = TextFieldProps & {
   name: string;
   native?: boolean;
   maxHeight?: boolean | number;
-  children: React.ReactNode;
+  children?: React.ReactNode;
+  options?: SelectOption[];
+  leftIcons?: ReactNode;
 };
 
 export function RHFSelect({
   name,
-
   helperText,
   children,
+  options = [],
+  leftIcons,
   ...other
 }: RHFSelectProps) {
   const { control } = useFormContext();
 
+  const getLeftIcons = (): Maybe<ReactNode> => {
+    if (leftIcons === undefined) {
+      return undefined;
+    }
+
+    return (
+      <Box display="flex" alignItems="center">
+        {leftIcons && (
+          <InputAdornment position="start">
+            {leftIcons}
+            <Divider sx={{ height: 28, m: 0.5 }} orientation="vertical" />
+          </InputAdornment>
+        )}
+      </Box>
+    );
+  };
+
```

Comment:
This is not really reactish. I just wonder what is the impact of this syntax about rendering optimization achieved by React as in this case we are recreating this function at each render... Does React rerender this component at each render even if this is not useful ?